pallets/wnodl/src/lib.rs

```diff
@@ -0,0 +1,197 @@
+    /// This would make sense to create a custom rpc and an off-chain storage for this purpose later.
+    pub type Balances<T: Config> =
+        StorageMap<_, Twox64Concat, T::AccountId, (BalanceOf<T>, BalanceOf<T>)>;
+
+    #[cfg(feature = "runtime-benchmarks")]
+    #[pallet::storage]
+    #[pallet::getter(fn benchmark_known_customers)]
+    /// An internally kept list for the benchmark tests.
+    pub type WhitelistedCallers<T: Config> = StorageValue<_, Vec<T::AccountId>>;
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Wrapping Nodl is initiated
+        /// parameters. [account's address on Nodle chain, amount of Nodl fund, destination address on Ethereum main-net]
+        WrappingInitiated(T::AccountId, BalanceOf<T>, EthAddress),
+
+        /// Wrapping Nodl is settles
+        /// parameters. [account's address on Nodle chain, amount of Nodl fund settled, Transaction hash on Ethereum main-net, reporting oracle's address]
+        WrappingSettled(T::AccountId, BalanceOf<T>, EthTxHash),
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// The balance of the account is not sufficient for the requested transaction
+        BalanceNotEnough,
+        /// The amount of fund to wrap should be between the pre-specified limits for the pallet.
+        FundNotWithinLimits,
+        /// We do not expect this error ever happen. But if it happened we would not allow it to mess with the correctness of the storage.
+        BalanceOverflow,
+        /// The customer is not known, whitelisted for this operation.
+        NotEligible,
+        /// Settlment is only possible for an amount equal or smaller than an initiated wrapping for a known customer
+        InvalidSettle,
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Initiate wrapping an amount of Nodl into wnodl on Ethereum
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        pub fn initiate_wrapping(
+            origin: OriginFor<T>,
+            amount: BalanceOf<T>,
+            eth_dest: EthAddress,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            #[cfg(feature = "runtime-benchmarks")]
+            if let Some(whitelisted_callers) = WhitelistedCallers::<T>::get() {
+                ensure!(whitelisted_callers.contains(&who), Error::<T>::NotEligible);
+            } else {
+                ensure!(T::KnownCustomers::contains(&who), Error::<T>::NotEligible);
+            }
+            #[cfg(not(feature = "runtime-benchmarks"))]
+            ensure!(T::KnownCustomers::contains(&who), Error::<T>::NotEligible);
+
+            let free_balance = T::Currency::free_balance(&who);
+            let minimum_balance = T::Currency::minimum_balance();
+            ensure!(
+                free_balance >= amount.saturating_add(minimum_balance),
+                Error::<T>::BalanceNotEnough
+            );
```

Comment:
I think we could let them go under the minimum balance if they want to move fully to eth

Reply:
Done